scripting/lua/api/ServerCb.cpp

```diff
@@ -43,9 +45,27 @@ const std::vector<ServerCbProxy::CustomRegType> ServerCbProxy::REGISTER_CUSTOM =
 		"commitPackage",
 		&ServerCbProxy::commitPackage,
 		false
+	},
+	{
+		"getRNG",
+		&ServerCbProxy::getRNG,
+		false
 	}
 };
 
+int ServerCbProxy::getRNG(lua_State * L)
+{
+	LuaStack S(L);
+	ServerCallback * obj = nullptr;
+
+	if(!S.tryGet(1, obj))
+		return S.retVoid();
+
+	S.clear();
+	S.push(dynamic_cast<CRandomGenerator *>(obj->getRNG()));
```

Comment:
it should be no cast here and API for RNG should use vstd::RNG. Also you should use `RNGMock` in tests

Reply:
I can try to move nextInt and nextDouble from CRandomGenerator to vstd::RNG and use vstd::RNG* in API.

Reply:
Please do not move, just declare as pure virtual

Reply:
Fixed this. Was not able to set mocked RNG for VR:T.